valiu-node-rpc/tests/integration_tests/transfer.rs

```diff
@@ -1,24 +1,16 @@
-use sp_keyring::AccountKeyring;
-use substrate_subxt::{extrinsic::PairSigner, sudo::SudoCallExt, ClientBuilder};
-use valiu_node_rpc::{TransferCall, ValiuRuntime};
+use sp_core::sr25519::Pair;
+use substrate_subxt::{extrinsic::PairSigner, Client};
+use valiu_node_rpc::{TransferCallExt, ValiuRuntime};
 
-#[tokio::test]
-async fn transfer() {
-    let _ = env_logger::builder().is_test(true).try_init();
-
-    let alice_pair = PairSigner::new(AccountKeyring::Alice.pair());
-    let bob_pair = PairSigner::new(AccountKeyring::Alice.pair());
-
-    let mut client = ClientBuilder::<ValiuRuntime>::new().build().await.unwrap();
-
-    let _ = crate::add_bob_as_a_member(&mut client, &alice_pair).await;
-
-    let encoded = client
-        .encode(TransferCall {
-            to: AccountKeyring::Bob.public(),
-            to_amount: 0,
-        })
+pub async fn transfer(
+    c: &mut Client<ValiuRuntime>,
+    alice_pair: &PairSigner<ValiuRuntime, Pair>,
+    bob_pair: &PairSigner<ValiuRuntime, Pair>,
+) {
+    use sp_core::Pair;
+    let rslt = c
+        .transfer_and_watch(bob_pair, alice_pair.signer().public(), 0)
```

Comment:
The order of this method is already the same of 

https://github.com/valibre-org/vln-node/blob/cad650b5ba29cefc79672236163220fad767203f/pallets/liquidity-provider/src/lib.rs#L127-L131

Yes, the method succeeded because the node emitted an event and didn't return an error. IMO, these tests should only verify connection related stuff and business logic (like `0` transfer) should be handled inside `pallets/liquidity-provider/tests.rs`

Reply:
Ok sounds good, but out of curiosity what kind of errors would be returned when doing the call? I'm still not familiar with the responses from submitting extrinsics, is it network/high level errors or would a validation error in the dispatchable call(like checking the amount is > 0) be also present in the `rslt`? other a bit off-topic question, when I get the result is it immediate  before the block was validated right? do I get the validated state as an event as well?

Reply:
> Ok sounds good, but out of curiosity what kind of errors would be returned when doing the call?

Most things are related to https://docs.rs/substrate-subxt/0.13.0/substrate_subxt/enum.Error.html. For example, decoding errors related to `substrate_subxt::ExtrinsicSuccess`, the returning type of all external calls; bad expected types between runtimes; corrupted node storage; etc. Validation stuff should probably be inside `Error::Invalid` or `Error::Other`.

> when I get the result is it immediate before the block was validated right? do I get the validated state as an event as well?

There are basically three external variants: (1) `submit` sends the encoded extrinsic to the blockchain and returns its hash. (2) `watch` awaits the block finalization. (3) `submit_and_watch` do both things sequentially.
`transfer` and `transfer_and_watch` are just specialized methods for the `transfer` call.

Reply:
Cool thanks for the info, also was a bit confused by the returned hash of the `submit` thinking it was the hash of the finalized block which made me think the call hangs until the block was validated. Is the hash of the extrinsic useful in any way? can I use it to later get updates, the status or something?

Reply:
I am not aware of any method of `substrate_subxt` that directly uses an extrinsic hash as a function paramenter. However, some workflows interact with the `EventSubscription` struct -> https://github.com/paritytech/substrate-subxt/blob/5595b1e82f7bbd1ad83eb603691e98888ff26630/src/frame/balances.rs#L217-L221